### PR TITLE
Fixes pickup animation layering and colored chairs

### PR DESCRIPTION
**Pull request by alsoandanswer**
- bugfix: "Fixes pickup animations not applying overlays and not layering over mobs."
  - bugfix: "Fixes colored chairs not re-applying colors when picked up and placed down."

### CODE REVIEW

**Comment by alsoandanswer** [comment]
!bugfix

**Comment by alsoandanswer** [comment]
!review

File: code/game/objects/items/weapons/material/twohanded.dm [hunk 1]
```diff
@@ -23,7 +23,7 @@
 	var/force_unwielded
 	var/wield_sound = /decl/sound_category/generic_wield_sound
 	var/unwield_sound = null
-	var/base_icon
+//	var/base_icon //moved to items.dm
```
Comment (NonQueueingMatt) [inline code/game/objects/items/weapons/material/twohanded.dm:26]: Delete instead of commenting.

**Review by NonQueueingMatt — CHANGES_REQUESTED** [review]

File: code/game/objects/items/weapons/material/twohanded.dm (hunk 1 above)
Comment (alsoandanswer) [inline code/game/objects/items/weapons/material/twohanded.dm:26]:
```suggestion
```

File: code/game/atoms_movable.dm [hunk 2]
```diff
@@ -372,8 +372,12 @@
 /atom/movable/proc/do_pickup_animation(atom/target)
 	if(!isturf(loc))
 		return
-	var/image/pickup_animation = image(icon, loc, icon_state, layer + 0.1, dir, pixel_x, pixel_y)
+	var/image/pickup_animation = image(icon, loc, icon_state, ABOVE_ALL_MOB_LAYER, dir, pixel_x, pixel_y)
 	pickup_animation.color = color
+	if(isitem(src))
+		var/obj/item/I = src
+		I.update_icon()
+		pickup_animation.overlays = I.overlays
```
Comment (NonQueueingMatt) [inline code/game/atoms_movable.dm:380]: No istype(src, ...) checks. Pass the overlays to the atom/movable proc through an override on the item level.

**Review by NonQueueingMatt — CHANGES_REQUESTED** [review]

**Pull request merged**